include/xtensor/xsort.hpp

```diff
@@ -497,60 +557,20 @@ namespace xt
     }
 
     template <class E, class C, class = std::enable_if_t<!xtl::is_integral<C>::value, int>>
-    inline auto partition(const xexpression<E>& e, const C& kth_container, std::ptrdiff_t axis = -1)
+    inline auto partition(const xexpression<E>& e, C kth_container, std::ptrdiff_t axis = -1)
```

Comment:
@JohanMabille Actually a copy of the `kth` container needs to be made in order to sort it so perhaps better to go with a universal reference.

Reply:
My point is, if you need a copy of `kth_container` in the implementation of `partition`, let's pass this argument by value so that the copy is not hidden to the user.